pan-domain-auth-play/src/main/scala/com/gu/pandomainauth/service/OAuth.scala

```diff
@@ -45,7 +46,7 @@ class OAuth(config: OAuthSettings, system: String, redirectUrl: String) {
   }
 
   def redirectToOAuthProvider(antiForgeryToken: String, email: Option[String] = None)
-                      (implicit context: ExecutionContext, request: RequestHeader, ws: WSClient): Future[Result] = {
+                      (implicit context: ExecutionContext): Future[Result] = {
```

Comment:
It isn't immediately obvious to me why you have moved these implicit parameters to the class level

Reply:
They're not strictly necessary, but it made life a bunch easier to move the implicits for the `discoveryDocument` method up to be shared throughout the class, so that I can make the initial discovery doc request at construction. Once they're up at the class level, these implicits become unnecessary, because the values used in the construction of the OAuth class are also the values that are passed to `redirectToOAuthProvider`.